api/src/exam-environment/routes/exam-environment.test.ts

```diff
@@ -0,0 +1,601 @@
+          data: newGeneratedExam
+        });
+
+        const body3: Static<typeof examEnvironmentPostExamGeneratedExam.body> =
+          {
+            examId: mock.examId
+          };
+
+        const res3 = await superPost('/exam-environment/exam/generated-exam')
+          .send(body3)
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          );
+
+        // Time is greater than 24 hours. So, request should pass, and new exam should be generated
+        expect(res3).toMatchObject({
+          status: 200,
+          body: {
+            data: {
+              examAttempt: {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                id: expect.not.stringMatching(mock.examAttempt.id)
+              }
+            }
+          }
+        });
+      });
+
+      it('should return the current attempt if it is still ongoing', async () => {
+        const latestAttempt =
+          await fastifyTestInstance.prisma.envExamAttempt.create({
+            data: mock.examAttempt
+          });
+
+        const body: Static<typeof examEnvironmentPostExamGeneratedExam.body> = {
+          examId: mock.examId
+        };
+
+        const res = await superPost('/exam-environment/exam/generated-exam')
+          .send(body)
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          );
+
+        expect(res).toMatchObject({
+          status: 200,
+          body: {
+            data: {
+              examAttempt: latestAttempt
+            }
+          }
+        });
+      });
+
+      it('should return an error if Camper has taken all generated exams', async () => {
+        // Add completed attempt for generated exam
+        const submittedAttempt = structuredClone(mock.examAttempt);
+        // Long-enough ago to be considered "submitted", and not trigger cooldown
+        submittedAttempt.startTimeInMS =
+          Date.now() -
+          24 * 60 * 60 * 1000 -
+          mock.exam.config.totalTimeInMS -
+          1 * 60 * 60 * 1000;
+        submittedAttempt.submissionTimeInMS =
+          Date.now() - mock.exam.config.totalTimeInMS - 24 * 60 * 60 * 1000;
+        await fastifyTestInstance.prisma.envExamAttempt.create({
+          data: submittedAttempt
+        });
+
+        const body: Static<typeof examEnvironmentPostExamGeneratedExam.body> = {
+          examId: mock.examId
+        };
+        const res = await superPost('/exam-environment/exam/generated-exam')
+          .send(body)
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          );
+
+        expect(res).toMatchObject({
+          status: 500,
+          body: {
+            code: 'FCC_ERR_EXAM_ENVIRONMENT'
+          }
+        });
+      });
+
+      it('should store an empty exam attempt in the database', async () => {
```

Comment:
Why is there an exam attempt in the db before anyone has attempted to do an exam?

Reply:
Thanks, that makes sense. How about this?
```suggestion
      it('should record the fact the user has started an exam', async () => {
```
I think it's useful to explain why we're doing this, rather than simply that we are.